bridge-history-api/internal/orm/batch_event.go

```diff
@@ -124,3 +125,15 @@ func (c *BatchEvent) UpdateBatchEventStatus(ctx context.Context, batchIndex uint
 	}
 	return nil
 }
+
+// GetLastFinalizedBatchIndex returns the last finalized batch index in db.
+func (c *BatchEvent) GetLastFinalizedBatchIndex(ctx context.Context) (uint64, error) {
+	var batche BatchEvent
+	db := c.db.WithContext(ctx)
+	db = db.Model(&BatchEvent{})
+	db = db.Select("batch_index")
+	db = db.Where("batch_status = ?", btypes.BatchStatusTypeFinalized)
+	db = db.Order("batch_index desc")
+	err := db.First(&batche).Error
```

Comment:
suggesting handling `not found` explicitly, and returning 0 here for readability.

Reply:
I have dealt with the `not found` error at the caller of this method.
Except the `not found` error, there may other error types which will are not easy to handle in `inner` method.
In this case, the caller will panic (log.Crit) when encountering other errors since there are not better way to handle it..
And it would be a little weird if the `inner` function only handles `not found` and return other type errors.
In conclusion I think this `inner` function should just return error and let the caller to deal with it.